src/main/java/Glen.java

```diff
@@ -23,7 +22,8 @@ public static void main(String[] args) {
                 String end = "";
                 try {
                     end = input.substring(++index);
-                } finally {
+                } catch (StringIndexOutOfBoundsException e) {}
```

Comment:
I think it's small detail but `catch (StringIndexOutOfBoundsException e) {}` should be replaced with
```Java
try {
    # Your remaining code here
} catch (StringIndexOutOfBoundsException e) {
} finally {
    # Your remaining code here
}